### ISSUE TITLE
Axis placement should accept CartesianIndex{2}

### ISSUE THREAD

**Issue by yakir12** [JuliaPlots/Makie.jl#1328]
So while this is possible:
```julia
fig = Figure()
Axis(fig[1, 1])
```
this is not:
```julia
fig = Figure()
i = CartesianIndex(1, 1)
Axis(fig[i])
```
In my opinion, this should be possible.

**Comment by jkrumbiegel** [comment]
Is there a specific reason why you need CartesianIndex? In reality the indices are ranges, so I'm not sure that CartesianIndices don't increase the misconceptions about GridLayouts being matrices.

**Comment by yakir12** [comment]
I agree that one can always avoid using the `LinearIndices` and `CartesianIndices` mechanisms, but it seems like an obvious use-case for them. I could add that `CartesianIndices` can be interpreted as index ranges as well. 

My main use-case is tiling a figure with `n` plots. I need to decide on the number of rows and columns the figure will have, and populate them:
```julia
sz = (9, 13)
fig = Figure()
for (i, row) in zip(CartesianIndices(sz), eachrow(dataframe))
  ax = Axis(fig[Tuple(i)...])
  lines!(row.points)
end
```
This touches a functionality similar to imagemagick's montage or a word-cloud where you know the figure's desired dimension and the total number of plots you have, and it finds the ideal nrows × mcols that best populates that figure. 

But regardless, CartesianIndex seems like a pretty natural connection to me. But as you've warned, I might be totally wrong.

**Comment by yakir12** [comment]
A very good tendency. Feel free to close this. This issue will be as useful closed as it would be open.
